### PR TITLE
Adding single-host integration test for GRPO and GSPO.

### PR DESCRIPTION
**Pull request by NicoGrande**
# Description

Adds an end-to-end integration test for both GRPO and GSPO workflows on qwen3-0.6b (which should work e2e on a single v6e-4 TPU VM). 

FIXES: b/490410666, b/490411100, b/490409949, b/490410743

# Tests

Tests are skipped for now pending debugging on v6e-4

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [x] I have performed a self-review of my code. For an optional AI review, add the `gemini-review` label.
- [x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [x] I have run end-to-end tests tests and provided workload links above if applicable.
- [x] I have made or will make corresponding changes to the doc if needed, including adding new documentation pages to the relevant Table of Contents (toctree directive) as explained in [our documentation](https://maxtext.readthedocs.io/en/latest/development.html#adding-new-documentation-files).

### CODE REVIEW

File: src/maxtext/configs/inference/vllm.yml [hunk 1]
```diff
@@ -14,6 +14,8 @@
 
 base_config: "base.yml"
 attention: "vllm_rpa"
+model_call_mode: "inference"
```
Comment (xuefgu) [inline src/maxtext/configs/inference/vllm.yml:17]: Where is this new config used below?

File: tests/integration/single_host_train_rl_test.py [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+        f"chips_per_vm={jax.device_count()}",
+        "scan_layers=True",
+        "hbm_utilization_vllm=0.4",
+        "rollout_data_parallelism=1",
+        "rollout_tensor_parallelism=-1",
+        "rl.num_generations=16",
+        "load_parameters_path=gs://maxtext-model-checkpoints/qwen3-0.6b/2025-10-27/scanned/0/items",
+    ]
+
+  def _run_rl_workflow_end_to_end(self, extra_argv):
```
Comment (xuefgu) [inline tests/integration/single_host_train_rl_test.py:86]: Why do we need this vs. directly calling `rl_train()` for various configs?

**Review by xuefgu — COMMENTED** [review]
Many thanks Nico! Just a couple of quick clarifications.

File: src/maxtext/configs/inference/vllm.yml (hunk 1 above)
Comment (NicoGrande) [reply src/maxtext/configs/inference/vllm.yml:17]: This is for MoE! Its unrelated but I wanted to squeeze it in to this PR since we should always be setting this in the vLLM codepath.

File: tests/integration/single_host_train_rl_test.py (hunk 2 above)
Comment (NicoGrande) [reply tests/integration/single_host_train_rl_test.py:86]: The biggest benefit is to be able to access some of the internals of `rl_train()`. For instance, this allows us to assert that the actor model params are being updated.

Comment (xuefgu) [reply tests/integration/single_host_train_rl_test.py:86]: The concern is the gradual/eventual divergence of logic in `rl_train()` and the tests. Any way we can mitigate that?

File: src/maxtext/configs/inference/vllm.yml (hunk 1 above)
Comment (dipannita08) [reply src/maxtext/configs/inference/vllm.yml:17]: Is the entry point for this through train_rl?

**Review by xuefgu — APPROVED** [review]
Deferring to you on how to handle the divergence between tests and application logic. Thanks again for the PR!

File: src/maxtext/configs/inference/vllm.yml (hunk 1 above)
Comment (NicoGrande) [reply src/maxtext/configs/inference/vllm.yml:17]: No, vllm.yml is only used in the MaxText on vLLM adapter class here: https://github.com/AI-Hypercomputer/maxtext/blob/main/src/maxtext/integration/vllm/maxtext_vllm_adapter/adapter.py#L73

File: tests/integration/single_host_train_rl_test.py (hunk 2 above)
Comment (NicoGrande) [reply tests/integration/single_host_train_rl_test.py:86]: Its a good point - I could add a couple of test cases that just calls `rl_train()` directly to test that API also.
